src/plugins/index_pattern_management/public/components/field_editor/field_editor.tsx

```diff
@@ -208,6 +212,7 @@ export class FieldEditor extends PureComponent<FieldEdiorProps, FieldEditorState
         data.fieldFormats
       ),
       fieldFormatId: get(indexPattern, ['fieldFormatMap', spec.name, 'type', 'id']),
+      displayName: get(indexPattern, ['attributes', 'fields', spec.name, 'displayName'], ''),
```

Comment:
agreed, will adapt and continue the suggestion you're provided tomorrow, thx!

Reply:
@mattkime I've merged your code, thx a lot, now I'd suggest to rename the persisted value to `customLabel`, since this is now part of fieldSpec, I think it's clearer this way ... customLabel persistence and types ... displayName ... the value that's actually displayed, which can be field.name oder field.customLabel. What do you think?

Reply:
💯 - good idea!